utils/pre-swap-checker/contracts/PreSwapChecker.sol

```diff
@@ -147,6 +149,64 @@ contract PreSwapChecker {
     return (errorCount, errors);
   }
 
+  /**
+    * @notice If order is going through wrapper to swap
+    * @param order Types.Order
+    * @param fromAddress address
+    * @param wrapper address
+    * @return uint256 errorCount if any
+    * @return bytes32[] memory array of error messages
+    */
+  function checkSwapWrapper(
+    Types.Order calldata order,
+    address fromAddress,
+    address payable wrapper
+    ) external view returns (uint256, bytes32[] memory ) {
+    IWETH wethContract =  Wrapper(wrapper).wethContract();
+    address swap = order.signature.validator;
+    // max size of the number of errors that could exist
+    bytes32[] memory errors = new bytes32[](4);
+    uint256 errorCount;
+
+    if (order.sender.wallet != fromAddress) {
+      errors[errorCount] = "MSG_SENDER_MUST_BE_ORDER_SENDER";
+      errorCount++;
+    }
+
+    // ensure that sender has approved wrapper contract on swap
+    if (!ISwap(swap).senderAuthorizations(order.sender.wallet, wrapper)) {
+      errors[errorCount] = "WRAPPER_UNAUTHORIZED";
```

Comment:
would it make more sense for the errors thrown here to match the errors the contracts would throw? SO its easy to see where they came from?

Reply:
ok.